client/src/components/GameTimer.jsx

```diff
@@ -1,23 +1,27 @@
 import { useState, useEffect } from 'react';
 
-export default function GameTimer({ duration }) {
-    const [secondsElapsed, setSecondsElapsed] = useState(Math.floor(duration / 1000));
+export default function GameTimer({ time }) {
+    const [secondsElapsed, setSecondsElapsed] = useState(
+        Math.floor(time / 1000)
+    );
 
     useEffect(() => {
         const timer = setInterval(() => {
-            setSecondsElapsed(prevSeconds => prevSeconds + 1);
+            setSecondsElapsed((prevSeconds) => prevSeconds + 1);
```

Comment:
This is risky. Incrementing the time every second relies on this code being executed exactly once per 1000ms. But the javascript environment can't promise you that. It will likely miss the mark by some milliseconds, especially if something else is happening that is heavy, causing the time to "drift". If the browser hangs for a few seconds, the timer will not increment and that time will be "lost".

A better way is always to store the start time (e.g. a `Date` object or the "Epoch time" when the game started) and then calculate the time since then. That way you only need the state to be stored in one place.

You can still force a re-render using state every second, but by setting the state to how many seconds have elapsed since start time.